code/game/gamemodes/wizard/rightandwrong.dm

```diff
@@ -33,16 +34,17 @@
 
 
 
-/mob/living/carbon/human/proc/equip_survivor(var/summon_type)
+/mob/living/carbon/human/proc/equip_survivor(var/datum/role/R)
+	var/summon_type = R.type
```

Comment:
Pretty sure that this will just return the typepath, and not `"magic"` or `"swords"`.
In other words, summon magic/summon swords is just going to give you guns.

Reply:
Ah, yes, good catch, I only tested guns. Got a fix here.